### PR TITLE
Feature/auth androidmanifest

### PR DESCRIPTION
**Pull request by yuqche**
Add script to unzip appcenter-loader-release.aar, modify the AndroidManifest, and zip back to add the section required for msal library (only for Mac Unity users)

Note: Still requires .NET 4.5 for current way of zipping and unzipping

[AB#68783](https://msmobilecenter.visualstudio.com/454a20a9-dfe6-4c1f-8ae8-417f76adb472/_workitems/edit/68783)

### CODE REVIEW

File: Assets/AppCenter/Editor/CreateManifest.cs [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+                    Arguments = $"-c \"cd {"appcenter-loader-release"} ; zip -r {"../Assets/AppCenter/Plugins/Android/appcenter-loader-release.aar"} {"*"} \"",
+                    RedirectStandardOutput = true,
+                    UseShellExecute = false,
+                    CreateNoWindow = true,
+                }
+            };
+            process.Start();
+            process.WaitForExit();
+        }
+        else
```
Comment (guperrot) [inline Assets/AppCenter/Editor/CreateManifest.cs]: remove else block, seems not used

File: Assets/AppCenter/Editor/CreateManifest.cs [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+#endif
+using UnityEditor.Build;
+using UnityEditor;
+using UnityEngine;
+
+public class CreateManifest
+{
+    public static void ZipFile(string sourceFile, string destinationFile)
+    {
+        if (Application.platform == RuntimePlatform.WindowsEditor)
```
Comment (guperrot) [inline Assets/AppCenter/Editor/CreateManifest.cs]: Maybe remove this if block since its empty

File: Assets/AppCenter/Editor/CreateManifest.cs [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+        {
+            UnityEngine.Debug.Log("entering else");
+        }
+    }
+
+    public static void UnzipFile(string sourceFile, string destinationFile)
+    {
+        if (Application.platform == RuntimePlatform.WindowsEditor)
+        {
+
```
Comment (guperrot) [inline Assets/AppCenter/Editor/CreateManifest.cs]: same here

File: Assets/AppCenter/Editor/CreateManifest.cs [hunk 4]
```diff
@@ -0,0 +1,148 @@
...
+        string manifestMetafile = "appcenter-loader-release/AndroidManifest.xml.meta";
+        
+        if (!File.Exists(loaderZipFile))
+        {
+            UnityEngine.Debug.LogWarning("Failed to load dependency file appcenter-loader-release.aar");
+            return;
+        }
+
+        // Delete unzipeed directory if it already exists.
+        if(Directory.Exists(loaderFolder))
```
Comment (guperrot) [inline Assets/AppCenter/Editor/CreateManifest.cs]: Reformat file

File: Assets/AppCenter/Editor/CreateManifest.cs (hunk 1 above)
Comment (yuqche) [reply Assets/AppCenter/Editor/CreateManifest.cs]: Sorry forgot to remove the debug message...

File: Assets/AppCenter/Editor/CreateManifest.cs (hunk 3 above)
Comment (yuqche) [reply Assets/AppCenter/Editor/CreateManifest.cs]: Added a TODO for this block as Windows Unity users do not have bash. May need other ways to run the unzip command on Windows machine

File: Assets/AppCenter/Editor/CreateManifest.cs [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+        var activityElements = applicationElements[0].Elements().Where(element => element.Name.LocalName == "activity").ToList();
+        
+        // Delete the unzipped folder if the activity element already exists in the AndroidManifest.xml file
+        if (activityElements.Count == 1) 
+        {
+            Directory.Delete(loaderFolder, true);
+            return;
+        }
+
+        var intentelement = new XElement("intent-filter");
```
Comment (owlstronaut) [inline Assets/AppCenter/Editor/CreateManifest.cs]: Looks like this variable should be `intentElement`?

**Review by owlstronaut — DISMISSED** [review]

File: Assets/AppCenter/Editor/CreateManifest.cs [hunk 6]
```diff
@@ -0,0 +1,186 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Xml.Linq;
+using UnityEngine;
+using System.Text;
```
Comment (dhei) [inline Assets/AppCenter/Editor/CreateManifest.cs]: sort using.

File: Assets/AppCenter/Editor/CreateManifest.cs [hunk 7]
```diff
@@ -0,0 +1,186 @@
...
+        }
+        if (processName.Length > 0) {
+            ExecuteProcess(processName, args);
+        }
+    }
+
+    public static void Create(AppCenterSettings settings)
+    {
+        var loaderZipFile = AppCenterSettingsContext.AppCenterPath + "/AppCenter/Plugins/Android/appcenter-loader-release.aar";
+        var loaderFolder = "appcenter-loader-release";
```
Comment (dhei) [inline Assets/AppCenter/Editor/CreateManifest.cs:114]: "appcenter-loader-release" is duplicated a few times. Can we extract it and use string.format() instead?

File: Assets/AppCenter/Editor/CreateManifest.cs [hunk 8]
```diff
@@ -0,0 +1,186 @@
...
+        var manifestPath = "appcenter-loader-release/AndroidManifest.xml";
+        var manifestMetafile = "appcenter-loader-release/AndroidManifest.xml.meta";
+
+        if (!File.Exists(loaderZipFile))
+        {
+            UnityEngine.Debug.LogWarning("Failed to load dependency file appcenter-loader-release.aar");
```
Comment (dhei) [inline Assets/AppCenter/Editor/CreateManifest.cs:120]: How about making "appcenter-loader-release.aar" a variable or constant?

File: Assets/AppCenter/Editor/CreateManifest.cs [hunk 9]
```diff
@@ -0,0 +1,186 @@
...
+        if (activityElements.Count == 1) 
+        {
+            Directory.Delete(loaderFolder, true);
+            return;
+        }
+
+        var intentelement = new XElement("intent-filter");
+        XNamespace ns = "http://schemas.android.com/apk/res/android";
+        var activityElement = new XElement("activity");
+        activityElement.SetAttributeValue(ns+"name", "com.microsoft.identity.client.BrowserTabActivity");
```
Comment (dhei) [inline Assets/AppCenter/Editor/CreateManifest.cs]:
```suggestion
        activityElement.SetAttributeValue(ns + "name", "com.microsoft.identity.client.BrowserTabActivity");
```
This line plus a few lines below.

File: Assets/AppCenter/Editor/CreateManifest.cs (hunk 7 above)
Comment (jaeklim) [reply Assets/AppCenter/Editor/CreateManifest.cs:114]: This is a temporary solution, we will revisit this one in the future.

File: Assets/AppCenter/Editor/CreateManifest.cs (hunk 8 above)
Comment (jaeklim) [reply Assets/AppCenter/Editor/CreateManifest.cs:120]: Mentioned above.

**Review by jaeklim — APPROVED** [review]

**Pull request merged**